src/extension_commands.ml

```diff
@@ -534,6 +534,215 @@ module Copy_type_under_cursor = struct
     command Extension_consts.Commands.copy_type_under_cursor handler
 end
 
+module Search_by_type = struct
+  let extension_name = "Search By Type"
+
+  let is_valid_file file_uri =
```

Comment:
@voodoos here we perform the check if it's a valid file
will this be sufficient for the purpose?

Reply:
I don't think checking the extension is a good idea. You should use the "language id" from the TextDocument that: https://vscode-api.js.org/interfaces/vscode.TextDocument.html#languageId

It's surprising that vscode doesn't always enforce the `when` clause in the package.json:
`"when": "editorLangId == ocaml || editorLangId == ocaml.interface || editorLangId == reason || editorLangId == ocaml.ocamllex"`